packages/react/menu/src/Menu.tsx

```diff
@@ -680,6 +718,7 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
           contentContext.onItemEnter(event);
           if (event.defaultPrevented) return;
           if (!props.disabled && !context.open && !openTimerRef.current) {
+            contentContext.onPointerGraceAreaChange(null);
```

Comment:
Why was this one needed?

Reply:
We need to clear the previous area when moving across triggers to prevent the old area from interfering with focus behaviour.